src/main/java/com/minecolonies/core/entity/ai/workers/AbstractEntityAIStructureWithWorkOrder.java

```diff
@@ -486,7 +486,7 @@ public void reduceNeededResources(final ItemStack stack)
     @Override
     protected boolean checkIfCanceled()
     {
-        if ((job.getWorkOrder() == null && job.getBlueprint() != null) || (structurePlacer != null && !structurePlacer.getB().hasBluePrint()))
+        if ((job.hasWorkOrder() && job.getWorkOrder() == null) || (job.getWorkOrder() == null && job.getBlueprint() != null) || (structurePlacer != null && !structurePlacer.getB().hasBluePrint()))
```

Comment:
has and get should be equvialent to get and null check, if has is true the get should always return sth

Reply:
hasWorkorder just checks if it has an id assigned atm, get actually fetches the workmanager one. I think it's also a performance thing. I could rename it to hasWorkOrderId or so